stdlib/public/core/KeyPath.swift

```diff
@@ -452,11 +458,15 @@ internal struct ComputedPropertyID: Hashable {
   @_inlineable // FIXME(sil-serialize-all)
   @_versioned // FIXME(sil-serialize-all)
   internal var hashValue: Int {
-    var hash = 0
-    hash ^= _mixInt(value)
-    hash ^= _mixInt(isStoredProperty ? 13 : 17)
-    hash ^= _mixInt(isTableOffset ? 19 : 23)
-    return hash
+    return _hashValue(for: self)
+  }
+
+  @_inlineable // FIXME(sil-serialize-all)
+  public func _hash(into hasher: _UnsafeHasher) -> _UnsafeHasher {
+    return hasher
+      .appending(value)
+      .appending(isStoredProperty)
+      .appending(isTableOffset)
```

Comment:
It would be nice, after the possibility of `inout _UnsafeHasher` and any coordinating changes to the API are sorted out, if in the final form of the API we could make a single call at the use site. That is, `hasher.append{ing}(foo, bar, baz)`.

Reply:
Can we do that for an arbitrary number of heterogeneous arguments?

Reply:
Ah right, they'd have to be `AnyHashable`-boxed. That said, if it's a transparent convenience function I'd expect the whole thing might be optimized away.

Reply:
I had some concerns about the original suggestion because of the fear that an array would need to be heap-allocated for the variable-length argument list. In some cases it looks like the compiler is smart enough to optimize that out—for example, unrolling a loop over those arguments. Is there any way we can guarantee that?